plugins/nf-lamin/src/main/nextflow/lamin/LaminObserver.groovy

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2025, Lamin Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.lamin
+
+import java.nio.file.Path
+import java.nio.file.PathMatcher
+import java.util.concurrent.locks.Lock
+import java.util.concurrent.locks.ReentrantLock
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+
+import nextflow.Session
+import nextflow.processor.TaskHandler
+import nextflow.processor.TaskRun
+import nextflow.trace.TraceObserver
+import nextflow.trace.TraceRecord
+
+import nextflow.lamin.api.Helper
+
+/**
+ * Example workflow events observer
+ *
+ * @author Robrecht Cannoodt <robrecht@data-intuitive.com>
+ */
+@Slf4j
+@CompileStatic
+class LaminObserver implements TraceObserver {
+    private Session session
+
+    private List<PathMatcher> matchers = []
+
+    private Set<TaskRun> tasks = []
+
+    private Set<Path> workflowInputs = []
+    private Map<Path,Path> workflowOutputs = [:]
+
+    private Lock lock = new ReentrantLock()
+
+    void logInfo(String message) {
+        log.info("nf-lamin> " + message)
+    }
+
+    @Override
+    void onFlowCreate(Session session) {
+        // store the session for later use
+        this.session = session
+
+        logInfo "onFlowCreate triggered!"
+
+        Helper.test()
+
+        def wfMetadata = session.getWorkflowMetadata()
+        def key = wfMetadata.scriptFile.toString().replaceFirst("${wfMetadata.projectDir}/", "")
+
+        // session.baseDir + "/" + session.scriptName
+
+        def description = session.config.navigate("manifest.description") as String
+
+        logInfo "Fetch or create Transform object:\n" +
+            "  trafo = ln.Transform(\n" +
+            "    key=\"${key}\",\n" +
+            "    version=\"${wfMetadata.revision}\",\n" +
+            "    type=\"pipeline\",\n" +
+            "    reference=\"${wfMetadata.repository}\",\n" +
+            "    reference_type=\"url\",\n" +
+            "    description=\"${wfMetadata.manifest.getDescription()}\"\n" +
+            "  )\n"
+
+        logInfo "Create Run object:\n" +
+            "  run = ln.Run(\n" +
+            "    transform=trafo,\n" +
+            "    name=\"${wfMetadata.runName}\",\n" +
+            "    started_at=\"${wfMetadata.start}\"\n" +
```

Comment:
Be careful with timezones 😅 I don't think the nf start time includes the timezone by default.

Reply:
Good point, TYSM!

Reply:
Just verified, `wfMetadata.start` contains the timezone:

    run = ln.Run(
      transform=trafo,
      name="thirsty_mendel",
      started_at="2025-04-08T11:18:08.935425391+02:00"
    )

so I think this should be fine. Thanks for getting me to verify this, though!